### ISSUE TITLE
Static check: ES6 classes must call super before referencing the "this" keyword

### ISSUE THREAD

**Issue by ChadKillingsworth** [google/closure-compiler#1160]
We should warn when an es6 class extends a parent, defines a constructor and references the `this` keyword before calling `super()`. By spec, the `this` keyword is null until the `super()` method is called.

See http://stackoverflow.com/a/31079103/1211524

```JavaScript
class foo extends bar {
  constructor() {
    this.foobar = 'foobar'; // null reference error - "this" is null before super.
    super();
  }
}
```

This also means we should warn when an arrow function is defined prior to `super()` calls if it references a `this` keyword.
